src/kmedoids_ucb.cpp

```diff
@@ -725,10 +784,16 @@ void KMedoids::swap(
         medoids.col(k) = data.col(medoid_indices(k));
         calc_best_distances_swap(
           data, medoid_indices, best_distances, second_distances, assignments);
+        arma::rowvec flat_sigma = sigma.as_row(); 
+        arma::rowvec P = {0.25, 0.5, 0.75};
+        arma::rowvec Q = arma::quantile(flat_sigma, P);
         std::ostringstream sigma_out;
-        sigma_out << "Sigma: min: " << sigma.min()
-        << ", max: " << sigma.max()
-        << ", mean: " << arma::mean(arma::mean(sigma));
+        sigma_out << "min: " << arma::min(flat_sigma)
+                  << ", 25th: " << Q(0)
+                  << ", median: " << Q(1)
+                  << ", 75th: " << Q(2)
+                  << ", max: " << arma::max(flat_sigma)
+                  << ", mean: " << arma::mean(flat_sigma);
```

Comment:
Instead of duplicating this code like in the `naive` algorithm, could you break this into a function that is called in both places?